src/site/topics/memory-callstack/memory-callstack.rst

```diff
@@ -118,81 +162,107 @@ Call Frame
         System.out.println("Finished function4");
     }
 
-* Running the above example would result in the call stack behaving like the below gif
 
-.. image:: callstack_animation.gif
-   :width: 250 px
-   :align: center
+* Running the above example would result in the call stack behaving like the below animation
 
+.. figure:: callstack_animation.gif
+    :width: 250 px
+    :align: center
+
+    Animation of call frames being pushed to and popped from the call stack.
 
 
 Heap
 ====
 
-* Very broadly speaking, the heap can be broken into static space and object space
+* Very broadly speaking, the heap can be broken into static space and instance space
 
 Static Space
 ------------
 
-* The static space contains one copy of all your code
+* The static space contains one copy of all the code
+
     * Classes
     * Static methods
     * Static variables
     * Interfaces
 
 
-Object Space
-------------
+Instance Space
+--------------
+
+* The instance space stores the object instances created at runtime
 
-* The object space stores the created objects
     * The object type
     * Values of the object's fields
+
         * This includes primitive types and references variables to other objects in the heap
 
-* When an object is created, it is put into the object part of the heap
+
+* When an object is created, it is put into the instance part of the heap
+
 
 
 Garbage Collection
 ==================
 
 * When a method finishes running and the call frame from the call stack is not needed anymore, the memory is deallocated
-* When an object in the object space is not needed anymore, *garbage collection* comes into play
+* When an object in the instance space is not needed anymore, *garbage collection* comes into play
+
+* In some programming languages, when something from the heap that has been allocated is not needed anymore, it must be manually deallocated
 
-* In some programming languages, when we don't need something from the heap that we've allocated anymore, we need to deallocate the memory it used
-    * If we don't we have a *memory leak*
+    * If it is not deallocated, it causes a *memory leak*
```

Comment:
Maybe worth mentioning, memory leaks can still happen in garbage collected languages. The easiest way is failing to close a file. (Look, java has destructors too.)